domain/resource/state/resource_test.go

```diff
@@ -795,14 +795,10 @@ func (s *resourceSuite) TestSetUnitResourceNotYetSuppliedExistingSupplierWrongTy
 
 	// Assert: an error is returned, nothing is updated in the db
 	c.Check(err, jc.ErrorIs, resourceerrors.UnknownRetrievedByType)
-	err = s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
-		var discard string
-		err = tx.QueryRow(`SELECT * FROM unit_resource`).Scan(&discard)
-		c.Check(err, jc.ErrorIs, sql.ErrNoRows, gc.Commentf("(Assert) unit_resource table has been updated: %v", errors.ErrorStack(err)))
-		err = tx.QueryRow(`SELECT * FROM resource_retrieved_by`).Scan(&discard)
-		c.Check(err, jc.ErrorIs, sql.ErrNoRows, gc.Commentf("(Assert) unit_resource table has been updated: %v", errors.ErrorStack(err)))
-		return nil
-	})
+	err = s.runQuery(`SELECT * FROM unit_resource`)
+	c.Check(err, jc.ErrorIs, sql.ErrNoRows, gc.Commentf("(Assert) unit_resource table has been updated: %v", errors.ErrorStack(err)))
+	err = s.runQuery(`SELECT * FROM resource_retrieved_by`)
+	c.Check(err, jc.ErrorIs, sql.ErrNoRows, gc.Commentf("(Assert) resource_retrieved_by table has been updated: %v", errors.ErrorStack(err)))
```

Comment:
I wonder if the query would have been better served as `SELECT COUNT(*) FROM resource_retrieved_by`, so you can count the number and assert the expectations. I'm not blocking the PR on this though. I just find it a bit odd.